### ISSUE TITLE
:Telescope repo list was unsuccessful - [fd error]

### ISSUE THREAD

**Issue by sgoudham** [cljoly/telescope-repo.nvim#43]
## The Issue
This appeared off the back of fixes deployed within the `dev` branch on the commit https://github.com/cljoly/telescope-repo.nvim/commit/b7fb147317ffb706215ef142b9ef4d6d81976015

When running the command `:Telescope repo list` and then `:checkhealth`, the following error is displayed

```shell
telescope._extensions.repo: require("telescope._extensions.repo.health").check()
========================================================================
  - ERROR: No markdown previewer found, the extension will not work properly
  - ERROR: No markdown previewer found, the extension will not work properly
  - ERROR: `cached_list` will not function without locate
  - OK: fd: found `fd`
    fd 8.4.0
  - ERROR: `:Telescope repo list` was unsuccessful. Exit code: 1
    fd --hidden --case-sensitive --absolute-path --exec echo {//} ; ^\.git$
    [fd error]: Command not found: Command { inner: "echo" "C:\\Users\\sgoud\\AppData\\Local\\nvim", remaining_argument_length: 8136 }
````

I tried to resolve this error myself and came across the following links which may be of use: 
- https://github.com/sharkdp/fd/issues/735
- https://github.com/sharkdp/fd/issues/722

I tried to get a version of the command `fd --hidden --case-sensitive --absolute-path --exec echo {//} ; ^\.git$` working locally but sadly could not.

Hope the above links help!

## Diagnostic Information
**nvim**: 0.7
**telescope**: [latest](https://github.com/nvim-telescope/telescope.nvim/commit/d3aad43b3fcf707052f7dd8a7c7072fa69773f3c)
**telescope-repo**: 0.3.0
**os**: Windows 11
**shell**: PowerShell

**Comment by sgoudham** [comment]
Stumbled upon this comment here while looking more into it -> https://github.com/sharkdp/fd/issues/538#issuecomment-591053535

Looks like the alias `echo` that's currently being used won't work within PowerShell nor will any other PowerShell-specific commands

I've had some success with the following 

```powershell
fd --hidden --case-sensitive --absolute-path '^\.git$' --exec powershell -Command echo `{//`}
```

but it's awfully slow and sometimes fails on repositories that it should recognise :/

**Comment by cljoly** [comment]
So to clarify, you got the plugin working on Windows using your print.bat script, right?

**Comment by sgoudham** [comment]
Yup, I did indeed. I had another idea that might be better but slower? We can guarantee that the ending of each file path will be `\.git\` so it might be possible to detect if the plugin is on windows and just remove the last 6 chars before passing it to Telescope?

**Comment by cljoly** [comment]
> Yup, I did indeed

Great! 🎉 

> So to clarify, you got the plugin working on Windows using your print.bat script, right?

In that case, we can add the bat script somewhere in the sources of the plugin and use it when we detect that we are on Windows.

> We can guarantee that the ending of each file path will be \.git\ so it might be possible to detect if the plugin is on windows and just remove the last 6 chars in lua before passing it to Telescope?

That seems worth looking into, plus Telescope offer functions to transform input as it’s displayed if I recall correctly. Thus, we might get decent performance even on huge list of repositories.

**Comment by sgoudham** [comment]
I would have ideally raised a PR by now but haven't had the motivation to go ahead and learn Lua syntax 😅 I think trimming the chars is a better solution globally as the regex guarantees the ending of the file paths. However, I feel like there's still room for edge cases... somewhere.

**Comment by cljoly** [comment]
There is a path module in plenary that could be of use.
I'll try to take a look later, but likely in 2 weeks.

**Comment by cljoly** [comment]
I’ve not been able to prioritize working on that yet.

**Comment by hinogi** [comment]
any progress on this?

**Comment by hinogi** [comment]
`fd --hidden --case-sensitive --absolute-path --exec powershell /C "echo {//}" ; ^\.git$` this seems to work on windows and powershell

https://github.com/sharkdp/fd/issues/538

**Comment by cljoly** [comment]
> any progress on this?


Your [PR](#53) would be a great solution! 😄
